### PR TITLE
perf(main): defer DB bootstrap, lazy-load heavy routers, minimal test defaults

### PR DESCRIPTION
**Pull request by jonpspri**
## Summary

Two related commits that together cut `import mcpgateway.main` cold-start time in test runs and drop the four slowest unit tests from 3-5s to under 0.5s.

**Commit 1 — `perf(main): defer DB bootstrap + lazy-load admin/rust proxy/llmchat`**

Several expensive operations happened at module-import time in `mcpgateway/main.py`, forcing every test that imported main (directly or via a fixture) to pay the full cost.

- Move `wait_for_db_ready()` and `bootstrap_db()` from module-level execution into the `lifespan()` startup hook. Module import no longer probes the database or runs Alembic.
- Drop the eager `from mcpgateway.admin import admin_router, set_logging_service` at the top of main. The 19k-line admin module is now imported lazily inside `if ADMIN_API_ENABLED:` along with `set_logging_service()`.
- Drop the eager `from mcpgateway.transports.rust_mcp_runtime_proxy import RustMCPRuntimeProxy`; import it inside `_build_mcp_transport_app()` on the branch that needs it.
- Gate `init_llmchat_redis()` in lifespan on `settings.llmchat_enabled` so disabled deployments skip the langchain import chain.

**Commit 2 — `test(conftest): disable admin/UI/LLM chat by default via reload fixture`**

With the lazy loading in place, the test harness can skip the admin module entirely for unit tests that don't exercise `/admin/...` routes. The tricky part: `mcpgateway.config.get_settings` is `@lru_cache`'d, so once main is imported with a given config, per-file env overrides can't flip the cached value back. This PR:

- Adds `_force_minimal_main_app_features()` to `tests/conftest.py` (runs at conftest import time, before any test collection). Forces `MCPGATEWAY_ADMIN_API_ENABLED=false`, `MCPGATEWAY_UI_ENABLED=false`, and `LLMCHAT_ENABLED=false` in the process env, clears `get_settings.cache_clear()`, and drops `mcpgateway.main` from `sys.modules` if it somehow snuck in.
- Adds a session-scoped `main_app_with_admin_api` fixture that clears the Settings lru_cache, strips shadowing instance attributes from the `LazySettingsWrapper` for the feature flags it cares about, pops `mcpgateway.main` from `sys.modules`, and reimports fresh with admin + UI flags flipped back on.
- Migrates the four existing unit test files that hit `/admin/...` via the real main.app (`test_well_known.py`, `test_admin_catalog_htmx.py`, `test_ui_version.py`, `test_final_coverage_push.py`) to pull the app from the new fixture instead of `from mcpgateway.main import app`.
- Flips `pyproject.toml` `[tool.pytest.ini_options].env` to `D:` defaults for the three feature flags, so host env can still override for manual runs.

**Key discovery:** `LazySettingsWrapper` has no `__setattr__` override, so `monkeypatch.setattr(settings, field, value)` sets direct instance attributes that shadow `__getattr__`. Pytest cleanup isn't always reliable for these shadowed attrs — several survive past their test's monkeypatch undo via the `_import_fresh_main_module` helper in `test_main_extended`. The fixture explicitly strips the feature flags from `settings.__dict__` before reloading main.

## Results

| metric | before | after |
|---|---|---|
| Cold `import mcpgateway.main` (test env) | ~1356ms | ~938ms |
| `test_main_registers_otel_request_middleware_when_tracing_is_enabled` | 5.20s | 0.34s |
| `TestErrorHandling::test_openapi_json_with_auth` | 3.70s | 0.25s |
| `test_module_level_reverse_proxy_router_success` | 3.00s | 0.24s |
| `test_module_level_uses_settings_backed_plugin_enablement` | 2.87s | 0.14s |

Pass count unchanged: **2905 passed / 5 failed** across the affected suites. The 5 failures are pre-existing caplog pollution caused by `test_main_extended.py`'s `_import_fresh_main_module` clobbering the root logger config — verified reproducible on `main` with `git stash`.

## Test plan

- [x] `uv run pytest tests/unit/mcpgateway/test_main.py` — 275 passed
- [x] `uv run pytest tests/unit/mcpgateway/test_main_extended.py` — all passed
- [x] `uv run pytest tests/unit/mcpgateway/test_well_known.py tests/unit/mcpgateway/test_admin_catalog_htmx.py tests/unit/mcpgateway/test_ui_version.py tests/unit/mcpgateway/test_final_coverage_push.py` — all admin-dependent tests pass via the new fixture
- [x] Full sweep of `tests/unit/mcpgateway/{test_main*, test_admin*, routers/}` — 2905 passed / 5 failed (matches `main` baseline)
- [x] Measured cold import time before/after
- [x] Verified the 4 previously-slow fresh-import tests drop from 3-5s to ~0.1-0.5s

### CODE REVIEW

File: mcpgateway/main.py [hunk 1]
```diff
@@ -1654,6 +1638,20 @@ async def lifespan(_app: FastAPI) -> AsyncIterator[None]:
...
+    # imported mcpgateway.main pay for a real DB probe and migration check.
+    # First-Party
+    from mcpgateway.bootstrap_db import main as bootstrap_db  # pylint: disable=import-outside-toplevel
+    from mcpgateway.utils.db_isready import wait_for_db_ready  # pylint: disable=import-outside-toplevel
+
+    wait_for_db_ready(
+        max_tries=int(settings.db_max_retries),
+        interval=int(settings.db_retry_interval_ms) / 1000,
+        sync=True,
+    )
```
[the remark this answers is not in the document]
Comment (jonpspri) [reply mcpgateway/main.py]: Valid — fixed in 4e0875b5a by wrapping the probe in `asyncio.to_thread`:

```python
await asyncio.to_thread(
    wait_for_db_ready,
    max_tries=int(settings.db_max_retries),
    interval=int(settings.db_retry_interval_ms) / 1000,
    sync=True,
)
```

Keeps the sync helper (no new async probe needed) while unblocking the event loop.

File: tests/conftest.py [hunk 2]
```diff
@@ -70,6 +67,43 @@ def _force_safe_test_db_defaults() -> None:
...
+    # drop it from sys.modules so the first genuine import picks up the
+    # minimal-feature settings.
+    sys.modules.pop("mcpgateway.main", None)
+
+
+_force_minimal_main_app_features()
+
 # First-Party
 import mcpgateway.db as db_mod  # noqa: E402  # must load after test DB env hardening
 from mcpgateway.config import Settings  # noqa: E402  # must load after test DB env hardening
```
[the remark this answers is not in the document]
Comment (jonpspri) [reply tests/conftest.py:113]: Acknowledged but not a regression introduced by this PR. `tests/conftest.py` already imported `mcpgateway.db` at module-import time on `origin/main` (that line predates this PR) and `mcpgateway.db` already reads `settings.database_url`, so the `get_settings` lru_cache was already populated before any test module was collected.

What this PR adds *helps* the scenario you describe: `_force_minimal_main_app_features()` calls `get_settings.cache_clear()` *after* setting the minimal-feature env vars, and drops `mcpgateway.main` from `sys.modules`. So test files that still rely on "set env before importing mcpgateway" get a freshly-rebuilt settings object on their first access.

The remaining edge case — a test file that both caches settings itself at collection time *and* depends on its own env vars winning over what conftest set — is not handled, but that pattern is exactly what the new `main_app_with_admin_api` fixture exists to replace. Marking resolved.

File: tests/conftest.py [hunk 3]
```diff
@@ -309,6 +343,56 @@ def app_with_temp_db():
...
+
+    for shadowed in ("mcpgateway_admin_api_enabled", "mcpgateway_ui_enabled", "llmchat_enabled"):
+        settings_wrapper.__dict__.pop(shadowed, None)
+
+    sys.modules.pop("mcpgateway.main", None)
+
+    # First-Party
+    from mcpgateway.main import app  # noqa: E402  # reload happens here
+
+    yield app
```
[the remark this answers is not in the document]
Comment (jonpspri) [reply tests/conftest.py]: Partially addressed in 4e0875b5a. The fixture now captures the prior `MCPGATEWAY_ADMIN_API_ENABLED` / `MCPGATEWAY_UI_ENABLED` env values and restores them on teardown.

I deliberately did **not** try to un-mount `admin_router` from `main.app`: FastAPI does not support reliable route removal, and the fixture is session-scoped, so any test that saw the mounted routes has already finished by the time teardown runs. Leaving the routes in place is also what the app would look like in production with admin enabled, so nothing should break.

The settings `get_settings` lru_cache is intentionally NOT cleared at teardown — clearing it while admin routes are still mounted would desync `settings.mcpgateway_admin_api_enabled` (false) from the live `main.app` routes (admin present), which is the worse of the two inconsistencies. Marking resolved.

File: pyproject.toml [hunk 4]
```diff
@@ -619,10 +619,19 @@ filterwarnings = [
...
-# Set environment variables for all tests
+# Set environment variables for all tests.
+# Heavy optional features (admin API, admin UI, LLM chat) are disabled by
+# default so `import mcpgateway.main` stays cheap. `tests/conftest.py` also
+# force-sets these at bootstrap time in case a host env flipped them on.
+# Tests that exercise the admin API via main.app use the
+# `main_app_with_admin_api` session fixture (tests/conftest.py), which
+# clears the Settings lru_cache and reloads mcpgateway.main with the flags
+# flipped back on. LLM chat unit tests import the router module directly,
+# so they do not need it mounted on main.app.
```
[the remark this answers is not in the document]
Comment (jonpspri) [reply pyproject.toml:630]: Valid — fixed in 4e0875b5a. `_force_minimal_main_app_features()` now uses `os.environ.setdefault(...)` instead of unconditional assignment:

```python
os.environ.setdefault("MCPGATEWAY_ADMIN_API_ENABLED", "false")
os.environ.setdefault("MCPGATEWAY_UI_ENABLED", "false")
os.environ.setdefault("LLMCHAT_ENABLED", "false")
```

A developer running `MCPGATEWAY_ADMIN_API_ENABLED=true pytest ...` to debug a single admin test now sees admin mounted at import time, matching the pyproject.toml comment. Marking resolved.
